packages/@react-spectrum/dialog/src/AlertDialog.tsx

```diff
@@ -84,24 +111,27 @@ function AlertDialog(props: SpectrumAlertDialogProps, ref: DOMRef) {
         {cancelLabel &&
           <Button
             variant="secondary"
-            onPress={() => chain(onClose(), onCancel())}
+            isDisabled={disabledByPending}
```

Comment:
In pending button we also set `aria-disabled` when it is pending but not yet visually disabled... getting kinda complicated to do from the outside.

How bad does it look if the secondary button immediately disables when pending?

Alternatively we could have a private prop or context for the "pending but not visually" state.

Are people gonna want to do this on other button groups too? e.g. what about in regular dialogs?

Reply:
IMO it doesn't look that bad to have the secondary button be immediately visually disabled when pending. However, the same kind of "flicker" that is described [here](https://spectrum.adobe.com/page/button/#Delay-before-pending-state) could happen then, just for the disabled state

> Alternatively we could have a private prop or context for the "pending but not visually" state.

I assume you meant "disabled but not visually" here? A private prop seems simple enough and would avoid the "flicker" that could happen. 

> Are people gonna want to do this on other button groups too? e.g. what about in regular dialogs?

Good point, I think they would. Maybe ButtonGroup could have have a prop "hasPending" or something that would handle setting the "disabled by not visually" prop onto the buttons via context and thus the 1 second timeout logic before becoming making the button fully disabled could live there? The actual pending Button would need to have `isPending` win against that "disabled by not visually" prop though 

However, if we might consider multiple buttons being pending at a single time in the future then might be best to keep this set at the Button level